spec/system/composer/prosemirror_editor_spec.rb

```diff
@@ -1140,4 +1140,148 @@ def body(title)
       )
     end
   end
+
+  describe "image toolbar" do
+    it "shows image toolbar when image is selected" do
+      cdp.allow_clipboard
+      open_composer_and_toggle_rich_editor
+      cdp.copy_test_image
+      cdp.paste
+
+      expect(rich).to have_css(".composer-image-node img")
+
+      rich.find(".composer-image-node img").click
+
+      expect(page).to have_css("[data-identifier='composer-image-toolbar']")
+      expect(page).to have_css(".composer-image-toolbar__zoom-out")
+      expect(page).to have_css(".composer-image-toolbar__zoom-in")
+      expect(page).to have_css(".composer-image-toolbar__trash")
+    end
+
+    it "allows scaling image down and up via toolbar" do
+      cdp.allow_clipboard
+      open_composer_and_toggle_rich_editor
+      cdp.copy_test_image
+      cdp.paste
+
+      rich.find(".composer-image-node img").click
+
+      find(".composer-image-toolbar__zoom-out").click
+
+      expect(rich.find(".composer-image-node img")["data-scale"]).to eq("75")
+
+      find(".composer-image-toolbar__zoom-out").click
+
+      expect(rich.find(".composer-image-node img")["data-scale"]).to eq("50")
+
+      find(".composer-image-toolbar__zoom-in").click
+
+      expect(rich.find(".composer-image-node img")["data-scale"]).to eq("75")
+
+      find(".composer-image-toolbar__zoom-in").click
+
+      expect(rich.find(".composer-image-node img")["data-scale"]).to eq("100")
+    end
+
+    it "allows removing image via toolbar" do
+      cdp.allow_clipboard
+      open_composer_and_toggle_rich_editor
+
+      composer.type_content("Before ")
+      cdp.copy_test_image
+      cdp.paste
+      composer.type_content(" After")
+
+      rich.find(".composer-image-node img").click
+
+      find(".composer-image-toolbar__trash").click
+
+      expect(rich).to have_no_css(".composer-image-node img")
+      expect(rich).to have_content("Before  After")
+    end
+
+    it "hides toolbar when clicking outside image" do
+      cdp.allow_clipboard
+      open_composer_and_toggle_rich_editor
+      cdp.copy_test_image
+      cdp.paste
+
+      rich.find(".composer-image-node img").click
+
+      expect(page).to have_css("[data-identifier='composer-image-toolbar']")
+
+      rich.find("p").click
+
+      expect(page).to have_no_css("[data-identifier='composer-image-toolbar']")
+    end
+  end
+
+  describe "image alt text display and editing" do
+    it "shows alt text input when image is selected" do
+      cdp.allow_clipboard
+      open_composer_and_toggle_rich_editor
+      cdp.copy_test_image
+      cdp.paste
+
+      rich.find(".composer-image-node img").click
```

Comment:
These lines are repeated a ton in this test, maybe could move them into `prepare_image_node` method?